plugins/forc-index/src/ops/forc_index_deploy.rs

```diff
@@ -41,6 +43,19 @@ pub async fn init(command: DeployCommand) -> anyhow::Result<()> {
         })?;
     }
 
+    // If we are replacing an indexer but not removing data, there is no need to
+    // issuer remove command. Ordinary reload is enough.
+    if replace_indexer && remove_data {
```

Comment:
- This here is the reason I don't think we _really_ need the complexity of a `kill_switch` in `handle_events`.
- A three second wait here would be more than enough time for the tokio task to stop 
  - Just now testing on my machine on origin/develop - the explorer never takes more than 2 seconds to stop after calling `forc index remove`
  - Remember `forc index deploy` is literally a deployment (it posts assets, registers them in the DB, creates a new task, starts that task -- it does a lot)
    - An additional small delay will not disrupt any UX here (especially if it means less complexity elsewhere)

Reply:
@lostman 👌🏽